code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -0,0 +1,354 @@
+		user.notransform = 1
+		user.revealed = 1
+		user.invisibility = 0
+		target.visible_message("<span class='warning'>[target] suddenly rises slightly into the air, their skin turning an ashy gray.</span>")
+		target.Beam(user,icon_state="drain_life",icon='icons/effects/effects.dmi',time=50)
+		target.notransform = 0
+		target.death(0)
+		target.visible_message("<span class='warning'>[target] gently slumps back onto the ground.</span>")
+		essence_drained = essence_drained * 5
+		user.essence += essence_drained
+		user.icon_state = "wraith"
+		user << "<span class='info'>Gained [essence_drained]E from [target]'s soul.</span>"
+		user << "<span class='info'>[target]'s soul has been all but destroyed. It will take some time for it to recuperate.</span>"
+		user.revealed = 0
+		user.notransform = 0
+		user.invisibility = INVISIBILITY_OBSERVER
+		drained_mobs.Add(target)
+		draining = 0
+
+
+//Transmit: the revemant's only direct way to communicate. Sends a single message silently to a single mob for 5E.
+/obj/effect/proc_holder/spell/targeted/revenant_transmit
+	name = "Unlock: Transmit (5E)"
+	desc = "Telepathically transmits a message to the target."
+	panel = "Revenant Abilities"
+	charge_max = 50
+	clothes_req = 0
+	range = 7
+	var/locked = 1
+
+/obj/effect/proc_holder/spell/targeted/revenant_transmit/cast(list/targets)
+	if(locked && essence_check(5, 1))
+		usr << "<span class='info'>You have unlocked Transmit!</span>"
+		name = "Transmit (5E)"
+		locked = 0
+		charge_counter = charge_max
+		return
+	if(locked)
+		charge_counter = charge_max
+		return
+	if(!essence_check(5))
+		return
+	for(var/mob/living/M in targets)
+		var/msg = stripped_input(usr, "What do you wish to tell \the [M.name]?.", null, "")
+		usr << "<span class='info'><b>You transmit to [M.name]:</b> [msg]</span>"
+		M << "<span class='deadsay'><b>Suddenly a strange voice resonates in your head...</b></span><i> [msg]</I>"
+
+
+//Fling Objects: Throws all items at the target. 20E
+/obj/effect/proc_holder/spell/targeted/revenant_throw
+	name = "Unlock: Fling Objects (20E)"
+	desc = "Throws all nearby loose objects at a single target."
+	panel = "Revenant Abilities"
+	charge_max = 200 //Short cooldown
+	clothes_req = 0
+	range = 5
+	var/locked = 1
+
+/obj/effect/proc_holder/spell/targeted/revenant_throw/cast(list/targets, var/mob/living/simple_animal/revenant/user = usr)
+	if(locked && essence_check(20, 1))
+		user << "<span class='info'>You have unlocked Fling Objects!</span>"
+		name = "Fling Objects (20E)"
+		locked = 0
+		charge_counter = charge_max
+		return
+	if(locked)
+		charge_counter = charge_max
+		return
+	if(!essence_check(20))
+		return
+	for(var/mob/living/M in targets)
+		var/list/objects_to_throw = list()
+		M << "<span class='warning'>You feel like a target...</span>"
+		for(var/obj/item/O in orange(3, M))
+			if(istype(O, /obj/item/device/radio/intercom))
+				continue
+			objects_to_throw.Add(O)
+		for(var/obj/item/O in orange(3, M))
```

Comment:
surely just loop the ```objects_to_throw``` list?

Also, consider ```spawn()```ing this and some of the other effects off, so that they happen simultaneously instead of one by one.